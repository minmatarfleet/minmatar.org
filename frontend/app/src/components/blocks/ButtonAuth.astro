---
import { i18n } from '@helpers/i18n'
const { t, translatePath } = i18n(Astro.url)

import type { ButtonColors, ButtonSizes } from '@dtypes/layout_components';

interface Props {
    color?:         ButtonColors;
    size?:          ButtonSizes;
    narrow?:        boolean;
    dialog_title:   string;
}

const {
    color = 'fleet-red',
    size = 'lg',
    narrow = false,
    dialog_title,
} = Astro.props;

const AUTH_REDIRECT_URL = `${translatePath('/redirects/auth_init')}?redirect_url=${Astro.url}`

import Button from '@components/blocks/Button.astro';
---

<Button
    type="button"
    color={color}
    size={size}
    narrow={narrow}
    x-data={`{
        disabled: false,
        show_auth_request_dialog() {
            this.disabled = true
            
            show_confirm_dialog({
                title: '${dialog_title}',
                partial: '${translatePath('/partials/authentication_missing_dialog/')}'
            }).then( (accepted) => {
                if (accepted)
                    window.location.href = '${AUTH_REDIRECT_URL}'
            })
        }
    }`}
    x-on:click="show_auth_request_dialog()"
    type="button"
>
    <slot />
</Button>