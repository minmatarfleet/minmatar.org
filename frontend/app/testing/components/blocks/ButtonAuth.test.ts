import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import ButtonAuth from "@components/blocks/ButtonAuth.astro";

test("ButtonAuth defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(ButtonAuth, {});

  expect(result).toMatchSnapshot();
});
