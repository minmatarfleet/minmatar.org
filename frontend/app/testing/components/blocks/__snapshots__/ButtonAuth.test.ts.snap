// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonAuth defaults 1`] = `
"<button type="button" x-data="{
        disabled: false,
        show_join_request_dialog() {
            this.disabled = true
            
            show_confirm_dialog({
                title: 'undefined',
                content: '<p>The following action requires authentication. A <a target=&#34;_blank&#34; rel=&#34;noreferrer noopener nofollow&#34; href=&#34;https://discord.com/invite/3hZfahmkFx&#34;>Minmatar Discord</a> user account is needed to continue. Would you like to proceed with authentication?</p>',
            }).then( (accepted) => {
                if (accepted)
                    window.location.href = '/redirects/auth_init?redirect_url=https://example.com/'
            })
        }
    }" x-on:click="show_join_request_dialog()" data-size="lg" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" disabled x-init="$el.removeAttribute(&#34;disabled&#34;)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">       </div>   </button> "
`;
